lib/krane/cli/global_deploy_command.rb

```diff
@@ -16,6 +16,9 @@ class GlobalDeployCommand
                              desc: "Verify workloads correctly deployed" },
         "selector" => { type: :string, banner: "'label=value'", required: true,
                         desc: "Select workloads owned by selector(s)" },
+        "selector-as-filter" => { type: :boolean,
+                                  desc: "Use --selector as a label filter to select a subset of resources to deploy",
```

Comment:
> A few small nits, overall the code looks fine. We should update the README to be a bit more explicit about this feature, since it's a bit of a special case whose use isn't immediately clear. In particular, we should be crystal clear about its intent (see readme suggestions). We should also explain that it preserves the pruning behaviour of `selector` as well.

Added some explanation in README `Sharing Namespace`.